rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/SSHInfrastructureV2.java

```diff
@@ -252,12 +261,15 @@ public void startNodeImpl(final InetAddress host, final int nbNodes, final List<
             Future<Void> deployResult = deployService.submit(new Callable<Void>() {
                 @Override
                 public Void call() throws Exception {
-                    while (!shutdown && !checkAllNodesAreAcquiredAndDo(createdNodeNames, null, null)) {
+                    while (!getShutdownFlag() && !checkAllNodesAreAcquiredAndDo(createdNodeNames, null, null)) {
                         if (anyTimedOut(depNodeURLs)) {
                             throw new IllegalStateException("The upper infrastructure has issued a timeout");
                         }
-                        if (chan.getExitStatus() != -1) { // -1 means process is
-                                                              // still running
+                        // we check the exit status of the session only in the
+                        // case where we link the current process to the one
+                        // that spawns the nodes. Otherwise, we let the two
+                        // processes live completely independently
+                        if (!deployNodesInDetachedMode && chan.getExitStatus() != -1) { // -1 means process is still running
```

Comment:
You could make this -1 a static variable in the class -> that would merge your comment and code

Reply:
addressed